src/org/omegat/core/machinetranslators/YandexCloudTranslate.java

```diff
@@ -115,14 +116,20 @@ public String getTranslation(Language sLang, Language tLang, String text, List<G
             return IAMErrorMessage;
         }
 
-        StringBuilder requestBuilder = new StringBuilder();
-        requestBuilder.append("{")
+        StringBuilder requestBuilder = new StringBuilder("{")
                 .append("\"sourceLanguageCode\":\"").append(sLang.getLanguageCode().toLowerCase()).append("\",")
                 .append("\"targetLanguageCode\":\"").append(tLang.getLanguageCode().toLowerCase()).append("\",")
                 .append("\"format\": \"HTML\",") // HTML format keeps OmegaT tags intact
-                .append("\"folderId\": \"").append(folderId).append("\",")
-                .append(getGlossaryConfigPart(glossaryTerms))
-                .append("\"texts\": [").append(JsonParser.quote(trText)).append("]}");
+                .append("\"folderId\": \"").append(folderId).append("\",");
+
+        if (!Preferences.isPreference(PROPERTY_USE_GLOSSARY) && glossarySupplier != null) {
```

Comment:
Please remove negation here. Otherwise all seems to work fine.

Reply:
Thanks! 🤦